gateway/ln-gateway/src/ng/pay.rs

```diff
@@ -319,6 +322,10 @@ impl GatewayPayInvoice {
             return Err(OutgoingContractError::TimeoutTooClose);
         }
 
+        if invoice.is_expired() {
+            return Err(OutgoingContractError::InvoiceExpired(invoice.expiry_time()));
+        }
```

Comment:
We might want the same on the client side. Maybe not that fn since it probably uses `SystemTime::now` which isn't WASM compatible, but the earlier we catch the problem the less hassle there is with refunding etc.